GameServerLib/Maps/MapScriptHandler.cs

```diff
@@ -0,0 +1,567 @@
+using System.Text;
+using Force.Crc32;
+using GameServerCore;
+using GameServerCore.Content;
+using GameServerCore.Domain;
+using GameServerCore.Domain.GameObjects;
+using GameServerCore.Enums;
+using GameServerCore.Maps;
+using LeagueSandbox.GameServer.Content;
+using LeagueSandbox.GameServer.GameObjects;
+using LeagueSandbox.GameServer.GameObjects.AttackableUnits.AI;
+using LeagueSandbox.GameServer.GameObjects.AttackableUnits.Buildings.AnimatedBuildings;
+using LeagueSandbox.GameServer.GameObjects.Other;
+using LeagueSandbox.GameServer.Logging;
+using LeagueSandbox.GameServer.Scripting.CSharp;
+using log4net;
+using MapScripts;
+
+namespace LeagueSandbox.GameServer.Maps
+{
+    /// <summary>
+    /// Class responsible for all map related game settings such as collision handler, navigation grid, announcer events, and map properties.
+    /// </summary>
+    public class MapScriptHandler : IMapScriptHandler
+    {
+        // Crucial Vars
+        protected Game _game;
+        public CSharpScriptEngine _scriptEngine;
+        public MapData _loadMapStructures;
+        private readonly ILog _logger;
+
+        /// <summary>
+        /// Unique identifier for the Map (ex: 1 = Old SR, 11 = New SR)
+        /// </summary>
+        public int Id { get; private set; }
+        /// <summary>
+        /// Collision Handler to be instanced by the map. Used for collisions between GameObjects or GameObjects and terrain.
+        /// </summary>
+        public ICollisionHandler CollisionHandler { get; private set; }
+        /// <summary>
+        /// Navigation Grid to be instanced by the map. Used for terrain data.
+        /// </summary>
+        public INavigationGrid NavigationGrid { get; private set; }
+        public IMapData MapData { get; private set; }
+        /// <summary>
+        /// MapProperties specific to a Map Id. Contains information about passive gold gen, lane minion spawns, experience to level, etc.
+        /// </summary>
+        public IMapScript MapScript { get; private set; }
+        /// <summary>
+        /// List of events related to the announcer (ex: first blood)
+        /// </summary>
+        public List<IAnnounce> AnnouncerEvents { get; private set; }
+
+        private int _minionNumber;
+        private int _cannonMinionCount;
+        public List<INexus> NexusList { get; set; } = new List<INexus>();
+        public Dictionary<string, IMapObject> SpawnBarracks { get; set; }
+        public Dictionary<TeamId, IFountain> FountainList { get; set; } = new Dictionary<TeamId, IFountain>();
+        private readonly Dictionary<TeamId, SurrenderHandler> _surrenders = new Dictionary<TeamId, SurrenderHandler>();
+        public Dictionary<TeamId, Dictionary<LaneID, List<IInhibitor>>> InhibitorList { get; set; }
+        public Dictionary<TeamId, Dictionary<LaneID, List<ILaneTurret>>> TurretList { get; set; }
+        public List<Tuple<IMinion, uint>> CapturePointsList { get; set; } = new List<Tuple<IMinion, uint>>();
```

Comment:
Change `CapturePointsList` to a `InfoPointList` of mapObjects. Usage of InfoPoints should be handled in-script.
```suggestion
        public List<IMapObject> InfoPoints { get; set; } = new List<IMapObject>();
```